cdap-runtime-ext-dataproc/pom.xml

```diff
@@ -135,6 +135,11 @@
       <groupId>org.powermock</groupId>
       <artifactId>powermock-api-mockito</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.cdap.cdap</groupId>
+      <artifactId>cdap-common</artifactId>
+      <version>${project.version}</version>
+    </dependency>
```

Comment:
extensions are not supposed to depend on `cdap-common`

Reply:
We are using the `Retries` package here which is implemented there.

Reply:
yeah I saw that but there are limitations we cannot use it.

Reply:
AFAIR, two main reasons for not using cdap-common: it's pretty big and you may face guava version collisions.

Reply:
Yeah, extensions should maintain the SPI boundary and be treated like they are completely separate from CDAP. In particular, the only CDAP dependency this should have is the `cdap-runtime-spi`.

Due to classloading and to avoid incompatible classes, we need to keep ext clean.

Reply:
Is there any way to use the logic here without the dependency? Will it be appropriate to create a class with a similar functionality in this module itself?

Reply:
Refactored the code as discussed offline